codespell_lib/_codespell.py

```diff
@@ -1096,6 +1103,88 @@ def flatten_clean_comma_separated_arguments(
     ]
 
 
+def get_git_executable_path() -> str:
+    # Find the full path to the git executable
+    git_path = shutil.which("git")
+    if git_path is None:
+        raise FileNotFoundError("Git executable not found in PATH")
+    return git_path
+
+
+def get_git_tracked_files(
+    root: str, files: Iterable[str], glob_match: GlobMatch, check_hidden: bool
+) -> Iterable[str]:
+    # Flatten the list of files into a single list of arguments for git ls-files
+    file_args = []
+    for filename in files:
+        if os.path.isdir(filename):
+            file_args.append(f"{filename}/**")
+        else:
+            file_args.append(filename)
+
+    # Add the glob patterns to exclude
+    exclude_patterns = [
+        f":(exclude)**/{pattern}" for pattern in glob_match.pattern_list
+    ]
+
+    # Add pattern to exclude hidden files if check_hidden is False
+    if not check_hidden:
+        exclude_patterns.append(":(exclude)**/.*")
+
+    git_executable = get_git_executable_path()
+
+    try:
+        result = subprocess.run(
+            [git_executable, "ls-files", *file_args, *exclude_patterns],
```

Comment:
For consistency, I wouldn't let `git` handle exclude patterns. I think the exclude patterns should be handled by the same codespell code, whatever the option. Said otherwise, I would create a list of tracked files, and let codespell do all the filtering stuff, based on that list and other exclude patterns.

I may be missing an obvious obstacle though.

Reply:
Letting git do the work here:
- Avoids having to create a separate git list, filter the already existing lisdt against that list.
- Letting git do it is faster IMHO as it uses its internal database, not the file system.  Walking throught the file system is slower.  Avoiding the extra filtering step is also a performance gain.